src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php

```diff
@@ -53,13 +48,16 @@ public function __construct($uploadPath, $segments, $logger = null)
     /**
      * {@inheritdoc}
      */
-    public function save($tempPath, $fileName, $version, $storageOption = null)
+    public function save($tempPath, $fileName, $preferredStorageOptions = null)
```

Comment:
This is not something new and I would keep it this way. Also the sharding is random so it could still be in the same folder.